query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusRequestHandlerBuilder.java

```diff
@@ -0,0 +1,39 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import javax.inject.Inject;
+import org.hypertrace.core.query.service.QueryServiceConfig.RequestHandlerClientConfig;
+import org.hypertrace.core.query.service.QueryServiceConfig.RequestHandlerConfig;
+import org.hypertrace.core.query.service.RequestHandler;
+import org.hypertrace.core.query.service.RequestHandlerBuilder;
+import org.hypertrace.core.query.service.RequestHandlerClientConfigRegistry;
+
+public class PrometheusRequestHandlerBuilder implements RequestHandlerBuilder {
+
+  private final RequestHandlerClientConfigRegistry clientConfigRegistry;
+
+  @Inject
+  PrometheusRequestHandlerBuilder(RequestHandlerClientConfigRegistry clientConfigRegistry) {
+    this.clientConfigRegistry = clientConfigRegistry;
+  }
+
+  @Override
+  public boolean canBuild(RequestHandlerConfig config) {
+    return "prometheus".equals(config.getType());
```

Comment:
nit : config.getType().equals("prometheus")

Reply:
I prefer this, just in case `config.getType` it still returns false instead of throwing exception